core/src/test/java/io/github/tcdl/msb/api/ChannelMonitorIT.java

```diff
@@ -37,10 +49,143 @@ public void testAnnouncement() throws InterruptedException {
+                .withBody(topicInfoMap)
+                .build();
+
+        CountDownLatch heartBeatResponseReceived = new CountDownLatch(1);
+        Callback<AggregatorStats> handler = stats -> {
+            assertTrue(stats.getTopicInfoMap().containsKey(TOPIC_NAME));
+            assertEquals(1, stats.getTopicInfoMap().get(TOPIC_NAME).getProducers().size());
+            heartBeatResponseReceived.countDown();
+        };
+
+        ChannelMonitorAggregator channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
+        channelMonitorAggregator.start(true, 10000, 1000);
+
+        //need to await for original request for heartbeat to be send to simulate response with same correlationId
+        Message requestMessage = awaitHeartBeatRequestSent(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME);
+
+        Message responseMessage = createMsbRequestMessage(requestMessage.getTopics().getResponse(), requestMessage.getCorrelationId(),
+                payload);
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(responseMessage, msbContext.getPayloadMapper()));
+
+        assertTrue("HeartBeat Response was not received",
+                heartBeatResponseReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
+    }
+
+    @Test
+    public void testHeartbeatUnexpectedMessage() throws InterruptedException {
+
+        CountDownLatch heartBeatResponseReceived = new CountDownLatch(1);
+        Callback<AggregatorStats> handler = stats -> {
+            assertEquals(0, stats.getTopicInfoMap().size());
+            heartBeatResponseReceived.countDown();
+        };
+
+        ChannelMonitorAggregator channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
+        channelMonitorAggregator.start(true, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, 1000);
+
+        //need to await for original request for heartbeat to be send to simulate response with same correlationId
+        Message requestMessage = awaitHeartBeatRequestSent(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME);
+
+        Message responseMessage = createMsbRequestMessage(requestMessage.getTopics().getResponse(), requestMessage.getCorrelationId(),
+                " unexpected statistics format received");
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(responseMessage, msbContext.getPayloadMapper()));
+
+        assertTrue("HeartBeat Response was not received",
+                heartBeatResponseReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
+    }
+
+    private Message awaitHeartBeatRequestSent(long timeout) {
+        long startTime = System.currentTimeMillis();
+        String originalRequest = null;
+        while (originalRequest == null && (System.currentTimeMillis() - startTime) < timeout) {
```

Comment:
i don't want to use consumer from msb here. Think about this just as overload for  MockAdapter.pollJsonMessageForTopic.  Move that to MockAdapter directly?.